server/testdriver/core/src/main/java/org/infinispan/server/test/core/ForkedServer.java

```diff
@@ -122,11 +144,35 @@ private void cleanServerLog() {
    }
 
    public File getServerLib() {
-      return Paths.get(serverHome + "/server/lib").toFile();
+      return Paths.get(serverHome + File.separator + "server" + File.separator + "lib").toFile();
    }
 
+   /**
+    * Returns the process ID of the Java process as opposed to the pid of the shell script which started the actual
+    * Java process. The process is matched containing the randomized UUID.
+    *
+    * @return process ID (pid) of the server's Java process.
+    */
    public long getPid() {
-      return org.infinispan.commons.jdkspecific.ProcessInfo.of(process).getPid();
+      try {
+         Process psProcess = Runtime.getRuntime().exec("ps");
```

Comment:
This won't work for Windows

Reply:
@tristantarrant None of this works on Windows (yet), please see FIXME on the top. It's next in my list, but we need to get this working at least on Linux.

Reply:
(@tristantarrant In case anyone is not familiar familiar with Windows the existing logic calling "java -9" doesn't work on Windows, it must eventually make use of "taskkill /f", etc.)